src/physics/em/BetheHeitlerInteractor.i.hh

```diff
@@ -0,0 +1,242 @@
+        else
+        {
+            // Used to sample from f2
+            epsilon = epsilon_min + (0.5 - epsilon_min) * rnd_b(rng);
+            // Calculate f2 density function (constant)
+            real_type f2 = 1 / (0.5 - epsilon_min);
+            // Calculate delta given the material Z and sampled epsilon
+            real_type d = this->delta(material_.Z(), epsilon);
+            // Calculate g2 "rejection" function
+            reject_threshold = this->Phi2_aux(d, material_.Z())
+                               / this->Phi2_aux(delta_min, material_.Z());
+        }
+        CHECK(epsilon >= epsilon_min && epsilon <= 0.5);
+    } while (generate_canonical(rng) > reject_threshold);
+
+    // Construct interaction for change to primary (incident) particle
+    Interaction result;
+    result.action      = Action::absorbed;
+    result.energy      = units::MevEnergy{0};
+    result.direction   = {0, 0, 0};
+    result.secondaries = {electron_pair, 2};
+
+    // Argument to density function for polar angle
+    real_type density_arg;
+    real_type log_r2r3
+        = std::log(generate_canonical(rng) * generate_canonical(rng));
+    if (generate_canonical(rng) < 0.25)
+    {
+        density_arg = log_r2r3 * 1.6;
+    }
+    else
+    {
+        density_arg = log_r2r3 / 1.875;
+    }
+
+    // Polar angle from the density function
+    real_type theta = 0.0977
+                      * (density_arg * std::exp(-log_r2r3)
+                         + 27 * density_arg * std::exp(-log_r2r3));
+
+    // Azimuthal angle sampled isotropically
+    UniformRealDistribution<real_type> sample_phi(0, 2 * constants::pi);
+    electron_pair[0].direction = rotate(
+        from_spherical(std::cos(theta), sample_phi(rng)), inc_direction_);
+    electron_pair[1].direction = rotate(
+        from_spherical(-std::cos(theta), sample_phi(rng)), inc_direction_);
+
+    // Outgoing secondaries are electron and positron
+    electron_pair[0].def_id = shared_.electron_id;
+    electron_pair[1].def_id = shared_.positron_id;
+    // Energies of secondaries shared equally
+    electron_pair[0].energy
+        = units::MevEnergy{0.5 * epsilon * inc_energy_.value()};
+    electron_pair[1].def_id
+        = units::MevEnergy{0.5 * epsilon * inc_energy_.value()};
+    // Calculate produced pair directions via conservation of momentum
+    for (unsigned int p = 0; p < allocate_.capacity(); p++)
+    {
+        for (int i = 0; i < 3; ++i)
+        {
+            electron_pair[p].direction[i]
+                = inc_direction_[i] * inc_energy_.value()
+                  - electron_pair[p].direction[i]
+                        * electron_pair[p].energy.value();
+        }
+        normalize_direction(&electron_pair[p].direction);
+    }
+
+    return result;
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::delta(size_type Z,
+                                                       real_type eps) const
+{
+    return 136 * std::pow(Z, -1 / 3) * epsilon0_ * epsilon0_
```

Comment:
Just as a note for later when we replace the mock material with the material track view and element view: there will be a `element.cbrt_z()` so you can can preferentially use that. There are a couple of other places that will need similar updates.